### PR TITLE
Fix k8s code generation

### PR DESCRIPTION
**Pull request by aanm**
Read per commit basis

### CODE REVIEW

File: contrib/scripts/check-k8s-code-gen.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ make generate-k8s-api manifests
 git add --intent-to-add .
 
 # Check for diff
-diff="$(git diff)"
+diff="$(git diff --staged)"
```
Comment (mhofstetter) [inline contrib/scripts/check-k8s-code-gen.sh]: we should add this here too: https://github.com/cilium/cilium/blob/main/contrib/scripts/check-api-code-gen.sh#L41

**Review by mhofstetter — APPROVED** [review]

File: contrib/scripts/check-api-code-gen.sh [hunk 2]
```diff
@@ -38,7 +38,7 @@ make generate-hubble-api
 git add --intent-to-add .
 
 # Check for diff
-diff="$(git diff)"
+diff="$(git diff --staged)"
```
Comment (bimmlerd) [inline contrib/scripts/check-api-code-gen.sh]: I think this breaks the detection of new files, again, though. Quick test repo, `existing` is a committed file, I've run `rm existing; echo "blub2" > new` and then `git add --intent-to-add .` . `git diff` shows the diff in `new`, `git diff --staged` shows the diff in `existing`, `git diff` shows the diff in `new`...

```
david@lynx:~/tmprepo:main -$ git add --intent-to-add .
david@lynx:~/tmprepo:main -$ git diff --staged        
diff --git a/existing b/existing
deleted file mode 100644
index 40142d0..0000000
--- a/existing
+++ /dev/null
@@ -1 +0,0 @@
-blub
david@lynx:~/tmprepo:main -$ git diff         
diff --git a/new b/new
new file mode 100644
index 0000000..4c30ff2
--- /dev/null
+++ b/new
@@ -0,0 +1 @@
+blub2
```

**Review by bimmlerd — CHANGES_REQUESTED** [review]

File: contrib/scripts/check-api-code-gen.sh (hunk 2 above)
Comment (mhofstetter) [reply contrib/scripts/check-api-code-gen.sh]: why using `--intent-to-add` at all? (only tested it with a normal `git add .` :man_facepalming: - then it works with `git diff --staged` for new files too (at least on my machine :P ))

**Review by bimmlerd — APPROVED** [review]
LGTM as is - as Marco H says, I'm not entirely sure what `intent-to-add` gives us, but Marco I will have had his reasons :)

**Pull request merged**